recipes/zstd/all/test_package/CMakeLists.txt

```diff
@@ -1,17 +1,13 @@
-cmake_minimum_required(VERSION 3.1)
-project(test_package C)
+cmake_minimum_required(VERSION 3.15)
```

Comment:
Could you try to decrease this version please? I don't think it's acceptable to fail in this case.

Reply:
The goal here is migration to Conan 2.0 which the agreed CMake version will be 3.15 [ref](https://github.com/conan-io/tribe/blob/main/design/004-tools-cmake.md) which is why it's being set here (and the other similar PRs)

Reply:
It's orthogonal. It's not because toolchain requires CMake >= 3.15 that it has not be compatible with CMakeLists files having 3.15 as minimum version.

Reply:
> Could you try to decrease this version please? I don't think it's acceptable to fail in this case. If it fails, it's a conan bug (if conan toolchain relies on policies >= 3.15, it should handle this policy management itself for a smooth user experience).

How is smoother for a user to fail later in the toolchain (because it will because it is checked) than failing in the `CMakeLists.txt`?  I would never say `I don't think it's acceptable` in this situation, this is the CMakeLists.txt of a test_package for the 2.X compatible build that for sure will require > 3.15. So I think we can totally live with a `cmake_minimum_required(VERSION 3.15)`, that is not a big deal at all.

Reply:
But requiring CMake >=3.15 doesn't mean than you can't be compatible with `cmake_minimum_required(VERSION 3.15)`, because anyway min version of most CMakeLists is lower.

Though, I've tested, and it's fine to decrease min version, conan injects the required policy externally.

Reply:
> But requiring CMake >=3.15 doesn't mean than you can't be compatible with cmake_minimum_required(VERSION 3.15). Min version of most CMakeLists of open source libs is lower.

Sure, but this is not any `open source library`, I'm ok not touching the `cmake_minimum_required` for the open source. This is the `CMakeLists.txt` for our test_package of conan 2.X.

Reply:
It should be noted that `cmake_minimum_required()` does _two_ things: stop  configuration if the version is lower, and set the baseline policy version. As you rightly point out, one can set specify policies without altering the cmake minimum required: we already do this in the generated toolchain files.

However, it should also be noted that our requirement to use CMake 3.15 goes _beyond_ just policy settings: the generated files by `CMakeToolchain` and `CMakeDeps` may and will use features that are only present starting in version 3.15.

These files do also check that the version is >= 3.15 and will fail then. My intention here is to simply "fail fast" - I understand this approach may be down to personal preference - if it doesn't "fail fast" during the `cmake_minimum_required()`, it will fail during the `project()` call when the project call is processed.

If one were to be strict and express a minimum required version as compatible with the package, the `TARGET_NAME_IF_EXISTS` generator expression, or the `TARGET_EXISTS` one in your suggestions, both require at least CMake 3.12.

Reply:
conan toolchain requires CMake >= 3.15, but in terms of policy relies on `CMP0091` set to `NEW`, and it doesn't expect any other policy, so it won't fail even with `cmake_minimum_required(VERSION 3.1)`

Reply:
For example in https://github.com/conan-io/conan-center-index/pull/12030, I've kept min version to 3.1, and it just works fine.